STM32/FLASH_readwrite/Src/FLASH_readwrite.c

```diff
@@ -88,72 +166,83 @@ uint32_t computeCRC(CRC_HandleTypeDef *hcrc, uint8_t *data, uint32_t size)
 
 /*!
 ** @brief Store data + computed CRC of the data in FLASH memory.
+** @return Returns 1 on error and 0 on success.
 */
-void writeToFlashCRC(CRC_HandleTypeDef *hcrc, uint32_t flash_address, uint8_t *data, uint32_t size)
+int writeToFlashCRC(CRC_HandleTypeDef *hcrc, uint32_t flash_address, uint8_t *data, uint32_t size)
```

Comment:
There is a lot of commonality between this function and writeToFlash. Seems like with a extra static function or two you could save some lines